packages/store-sync/src/dozer/fetchRecordsSql.ts

```diff
@@ -0,0 +1,69 @@
+import { DecodeDozerRecordsResult, DozerQueryResult, decodeDozerRecords } from "./decodeDozerRecords";
+import { Hex } from "viem";
+import { TableQuery } from "./common";
+import { Table } from "@latticexyz/config";
+
+type DozerResponseSuccess = {
+  block_height: string;
+  result: DozerQueryResult[];
+};
+
+type DozerResponseFail = { msg: string };
+
+type DozerResponse = DozerResponseSuccess | DozerResponseFail;
+
+function isDozerResponseFail(response: DozerResponse): response is DozerResponseFail {
+  return "msg" in response;
+}
+
+type FetchRecordsSqlArgs = {
+  dozerUrl: string;
+  storeAddress: Hex;
+  queries: TableQuery[];
+};
+
+type FetchRecordsSqlResult =
+  | {
+      blockHeight: bigint;
+      result: {
+        table: Table;
+        records: DecodeDozerRecordsResult;
+      }[];
+    }
+  | undefined;
+
+export async function fetchRecordsSql({
+  dozerUrl,
+  queries,
+  storeAddress,
+}: FetchRecordsSqlArgs): Promise<FetchRecordsSqlResult> {
+  const response: DozerResponse = await (
+    await fetch(dozerUrl, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(queries.map((query) => ({ address: storeAddress, query: query.sql }))),
+    })
+  ).json();
+
+  if (isDozerResponseFail(response)) {
+    console.warn(`Dozer response: ${response.msg}\n\nTry reproducing via cURL:
+    curl ${dozerUrl} \\
+    --compressed \\
+    -H 'Accept-Encoding: gzip' \\
+    -H 'Content-Type: application/json' \\
+    -d '[${queries.map((query) => `{"address": "${storeAddress}", "query": "${query.sql.replaceAll('"', '\\"')}"}`).join(",")}]'`);
+    return;
```

Comment:
I find it a little weird that a function to fetch a thing returns undefined on a failure rather than throwing or something.

I would usually reserve this behavior for the place where "failure is optional" e.g. `getSnapshot` or `syncToStash`

Reply:
no i think you're right, can't remember why i put it in here. moved it into `getSnapshot` now.